### ISSUE TITLE
Create independent documentation for `NNlib.jl`?

### ISSUE THREAD

**Issue by Saransh-cpp** [FluxML/NNlib.jl#430]
`NNlib.jl` does not have independent documentation, rather, it has a [dedicated page](https://fluxml.ai/Flux.jl/stable/models/nnlib/) in `Flux`'s documentation. Due to the lack of stand-alone documentation, it does not have doctests, and the documentation is not tested periodically.

Other repositories under `FluxML` have independent documentations, for example [Functors.jl](https://fluxml.ai/Functors.jl/stable/) and OneHotArrays.jl (deployment in progress (?)). Should `NNlib` also have such independent documentation to ensure that the docstrings and examples don't go stale?

**Comment by DhairyaLGandhi** [comment]
It would be good to add doctests (and set up Flux to run them as well), but by and large i think having a central place for documentation is good. It makes discovery easy

**Comment by darsnack** [comment]
Yes, NNlib should get its own documentation.

**Comment by CarloLucibello** [comment]
Agree, NNlib is used by different DL framework, it should have its own docs

**Comment by Saransh-cpp** [comment]
> but by and large i think having a central place for documentation is good. It makes discovery easy

Yes, it definitely makes the navigation and discoverability easy, but we could keep both of them - a dedicated section in `Flux` and stand-alone documentation! This would also not make `NNlib` the odd-one-out (concerning documentation) repository under `FluxML`.

I can create the basic structure of the documentation as a starting point and then build upon it iteratively with future PRs!

**Comment by Saransh-cpp** [comment]
Cross-posting this here from the PR -

---

The documentation should show up on the given link, but I think the GH pages are not enabled for `NNlib`? Could you please enable the pages from `NNlib`'s settings? Additionally, a [`DOCUMENTER_KEY`](https://juliadocs.github.io/Documenter.jl/stable/man/hosting/#Authentication:-SSH-Deploy-Keys) should be added as a repository secret in `NNlib.jl`.

Right now the deployment is failing -
```julia
┌ Info: Deployment criteria for deploying devbranch build from GitHub Actions:
│ - ✔ ENV["GITHUB_REPOSITORY"]="FluxML/NNlib.jl" occurs in repo="github.com/FluxML/NNlib.jl.git"
│ - ✔ ENV["GITHUB_EVENT_NAME"]="push" is "push"
│ - ✔ ENV["GITHUB_REF"] matches devbranch="master"
│ - ✔ ENV["GITHUB_ACTOR"] exists
│ - ✔ ENV["DOCUMENTER_KEY"] exists exists
└ Deploying: ✔
hint: Using 'master' as the name for the initial branch. This default branch name
hint: is subject to change. To configure the initial branch name to use in all
hint: of your new repositories, which will suppress this warning, call:
hint: 
hint: 	git config --global init.defaultBranch <name>
hint: 
hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
hint: 'development'. The just-created branch can be renamed via this command:
hint: 
hint: 	git branch -m <name>
Initialized empty Git repository in /tmp/jl_wxIzo9/.git/
Failed to add the ECDSA host key for IP address '192.30.255.112' to the list of known hosts (/home/runner/.ssh/known_hosts).
Load key "/home/runner/work/NNlib.jl/NNlib.jl/docs/.documenter": invalid format
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
┌ Error: Git failed to fetch git@github.com:FluxML/NNlib.jl.git
│ This can be caused by a DOCUMENTER_KEY variable that is not correctly set up.
│ Make sure that the environment variable is properly set up as a Base64-encoded string
│ of the SSH private key. You may need to re-generate the keys with DocumenterTools.
└ @ Documenter ~/.julia/packages/Documenter/PLD7m/src/Documenter.jl:502
┌ Error: Failed to push:
│   exception =
│    failed process: Process(`git fetch upstream`, ProcessExited(128)) [128]
│    
│    Stacktrace:
│      [1] pipeline_error
│        @ ./process.jl:538 [inlined]
│      [2] run(::Cmd; wait::Bool)
│        @ Base ./process.jl:453
│      [3] run
│        @ ./process.jl:451 [inlined]
│      [4] (::Documenter.var"#git_commands#22"{String, String, SubString{String}, String, Vector{Any}, Bool, String, String})(sshconfig::String)
│        @ Documenter ~/.julia/packages/Documenter/PLD7m/src/Documenter.jl:500
│      [5] (::Documenter.var"#21#27"{String})()
│        @ Documenter ~/.julia/packages/Documenter/PLD7m/src/Documenter.jl:598
│      [6] cd(f::Documenter.var"#21#27"{String}, dir::String)
│        @ Base.Filesystem ./file.jl:106
│      [7] #20
│        @ ~/.julia/packages/Documenter/PLD7m/src/Documenter.jl:598 [inlined]
│      [8] withenv(f::Documenter.var"#20#26"{String, String}, keyvals::Pair{String, String})
│        @ Base ./env.jl:161
│      [9] (::Documenter.var"#19#25"{String, String, SubString{String}})(sshconfig::String, io::IOStream)
│        @ Documenter ~/.julia/packages/Documenter/PLD7m/src/Documenter.jl:597
│     [10] mktemp(fn::Documenter.var"#19#25"{String, String, SubString{String}}, parent::String)
│        @ Base.Filesystem ./file.jl:[70](https://github.com/FluxML/NNlib.jl/runs/8074414339?check_suite_focus=true#step:6:71)3
│     [11] mktemp(fn::Function)
│        @ Base.Filesystem ./file.jl:701
│     [12] git_push(root::String, temp::String, repo::String; branch::String, dirname::String, target::String, sha::SubString{String}, devurl::String, versions::Vector{Any}, forcepush::Bool, deploy_config::Documenter.GitHubActions, subfolder::String)
│        @ Documenter ~/.julia/packages/Documenter/PLD7m/src/Documenter.jl:582
│     [13] #13
│        @ ~/.julia/packages/Documenter/PLD7m/src/Documenter.jl:456 [inlined]
│     [14] mktempdir(fn::Documenter.var"#13#15"{SubString{String}, String, String, String, String, String, String, Vector{Any}, Bool, Documenter.GitHubActions, String}, parent::String; prefix::String)
│        @ Base.Filesystem ./file.jl:[72](https://github.com/FluxML/NNlib.jl/runs/8074414339?check_suite_focus=true#step:6:73)9
│     [15] mktempdir(fn::Function, parent::String) (repeats 2 times)
│        @ Base.Filesystem ./file.jl:727
│     [16] (::Documenter.var"#12#14"{String, String, String, String, String, Nothing, String, Vector{Any}, Bool, Documenter.GitHubActions, String})()
│        @ Documenter ~/.julia/packages/Documenter/PLD7m/src/Documenter.jl:455
│     [17] cd(f::Documenter.var"#12#14"{String, String, String, String, String, Nothing, String, Vector{Any}, Bool, Documenter.GitHubActions, String}, dir::String)
│        @ Base.Filesystem ./file.jl:106
│     [18] deploydocs(; root::String, target::String, dirname::String, repo::String, branch::String, deps::Nothing, make::Nothing, devbranch::String, devurl::String, versions::Vector{Any}, forcepush::Bool, deploy_config::Documenter.GitHubActions, push_preview::Bool)
│        @ Documenter ~/.julia/packages/Documenter/PLD7m/src/Documenter.jl:432
│     [19] top-level scope
│        @ ~/work/NNlib.jl/NNlib.jl/docs/make.jl:17
│     [20] include(mod::Module, _path::String)
│        @ Base ./Base.jl:3[84](https://github.com/FluxML/NNlib.jl/runs/8074414339?check_suite_focus=true#step:6:85)
│     [21] exec_options(opts::Base.JLOptions)
│        @ Base ./client.jl:2[85](https://github.com/FluxML/NNlib.jl/runs/8074414339?check_suite_focus=true#step:6:86)
│     [22] _start()
│        @ Base ./client.jl:485
└ @ Documenter ~/.julia/packages/Documenter/PLD7m/src/Documenter.jl:603
```

**Comment by Saransh-cpp** [comment]
Closed by #431
